toolkit/tools/pkg/imagecustomizerlib/customizeboot.go

```diff
@@ -128,6 +178,31 @@ func setLinuxPath(inputGrubCfgContent string, linuxPath string) (outputGrubCfgCo
 	return outputGrubCfgContent, oldKernelPath, nil
 }
 
+// Overrides the path of the kernel binary of all the linux commands within a grub config file.
+func setLinuxPathAll(inputGrubCfgContent string, linuxPath string) (outputGrubCfgContent string, err error) {
+	quotedLinuxPath := grub.QuoteString(linuxPath)
+
+	lines, err := findLinuxLineAll(inputGrubCfgContent)
+	if err != nil {
+		return "", err
+	}
+
+	outputGrubCfgContent = inputGrubCfgContent
+	// loop from last to first so that the captured locations from
+	// findGrubCommandAll are not invalidated as reconstructing
+	// outputGrubCfgContent.
+	for i := len(lines); i > 0; i = i - 1 {
```

Comment:
`i` is an `int`. So, you can rewrite this as:

```golang
for i := len(lines)-1; i >= 0; i-- {
    line = lines[i]
```

Similar comment elsewhere.

Reply:
I switched away from this `for i := len(lines)` because `len(lines` can be 0 - and to protect against -1, I'd have to enclose the block with an `if len() > 0`. So, I opted to have `for i := len(lines); i > 0...` for more consise code. I'll update the `i--` part though.

Reply:
sigh, yes, of course...
pending

Reply:
fixed.